components/eamxx/scripts/test_all_scream.py

```diff
@@ -887,10 +888,8 @@ def generate_baselines(self, test, commit):
                     success = False
 
         finally:
-            # Clean up the directory, by removing everything but the 'data' subfolder. This must
-            # happen unconditionally or else subsequent runs could be corrupted
-            run_cmd_no_fail(r"find -maxdepth 1 -not -name data ! -path . -exec rm -rf {} \;",
-                            from_dir=test_dir, verbose=True, dry_run=self._dry_run)
+            # Clean up the directory by removing everything
+            run_cmd_no_fail(f"/bin/rm -rf {test_dir}/*")
```

Comment:
If I understand correctly, the location of SCREAM_DATA_DIR when generating baselines is in the baseline dir, while the actual code is built and run in the usual ctest-build subdir, right?

If that's the case, I would _not_ clean up the folder at the end. We already nuke them at the beginning of the test phase anyways. The advantage of not nuking them is that, if something fails, we can cd in the folder and see what was going on (e.g., inspect the cache, check flags, etc.).

Reply:
@bartgol , I had to move the "make test dirs and nuke them" call earlier in TAS in order to ensure this directory existed for generate_baseline. I don't think they get nuked again, so leaving the files around could case problems with the test build. That said, I don't think it would be hard to put the baseline build in a subdirectory that would be harmless to the main build. Something like `ctest-build/debug/baseline_build`.